### PR TITLE
【PIR API adaptor No.192】Migrate paddle.searchsorted into pir

### PR DESCRIPTION
**Pull request by Olive-2019**
### PR types

Others
### PR changes

APIs
### Description

https://github.com/PaddlePaddle/Paddle/issues/58067
PIR API 推全升级
将算子迁移升级至 pir，并更新单测
`paddle.searchsorted`（0/7）

### CODE REVIEW

**Comment by Olive-2019** [comment]
**失败单测**
1. test_check_output (__main__.TestSearchSorted)
2. test_static_api (__main__.TestSearchSortedAPI)
3. test_check_output (__main__.TestSearchSortedOp1)
4. test_check_output (__main__.TestSearchSortedOp2)
5. test_check_output (__main__.TestSearchSortedOp3)
6. test_check_output (__main__.TestSearchSortedOp4)
7. test_check_output (__main__.TestSearchSortedOp5)

**报错信息示例**
```
======================================================================
ERROR: test_check_output (__main__.TestSearchSorted)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_searchsorted_op.py", line 45, in test_check_output
    self.check_output(check_pir=True)
  File "/home/aistudio/Paddle/test/legacy_test/op_test.py", line 2762, in check_output
    check_auto_parallel=check_auto_parallel,
  File "/home/aistudio/Paddle/test/legacy_test/op_test.py", line 2622, in check_output_with_place
    pir_checker.check()
  File "/home/aistudio/Paddle/test/legacy_test/op_test.py", line 2187, in check
    self.calculate_output()
  File "/home/aistudio/Paddle/test/legacy_test/op_test.py", line 2373, in calculate_output
    pir_outs = self.op_test._calc_pir_output(place)
  File "/home/aistudio/Paddle/test/legacy_test/op_test.py", line 1379, in _calc_pir_output
    ret_tuple = self.python_api(*args)
  File "/home/aistudio/.local/lib/python3.7/site-packages/paddle/tensor/search.py", line 1157, in searchsorted
    out = helper.create_variable_for_type_inference(dtype=out_type)
  File "/home/aistudio/.local/lib/python3.7/site-packages/paddle/base/layer_helper_base.py", line 476, in create_variable_for_type_inference
    stop_gradient=stop_gradient,
  File "/home/aistudio/.local/lib/python3.7/site-packages/paddle/base/framework.py", line 4223, in create_var
    var = Variable(block=self, *args, **kwargs)
  File "/home/aistudio/.local/lib/python3.7/site-packages/paddle/base/framework.py", line 1514, in __init__
    self.desc.set_dtype(dtype)
TypeError: set_dtype(): incompatible function arguments. The following argument types are supported:
    1. (self: paddle.base.libpaddle.VarDesc, arg0: paddle::framework::proto::VarType_Type) -> None

Invoked with: <paddle.base.libpaddle.VarDesc object at 0x7f391f0a97f0>, <DataType.INT64: 9>
```

**Comment by MarioLulab** [comment]
> **失败单测**
> 
> 1. test_check_output (**main**.TestSearchSorted)
> 2. test_static_api (**main**.TestSearchSortedAPI)
> 3. test_check_output (**main**.TestSearchSortedOp1)
> 4. test_check_output (**main**.TestSearchSortedOp2)
> 5. test_check_output (**main**.TestSearchSortedOp3)
> 6. test_check_output (**main**.TestSearchSortedOp4)
> 7. test_check_output (**main**.TestSearchSortedOp5)
> 
> **报错信息示例**
> 
> ```
> ======================================================================
> ERROR: test_check_output (__main__.TestSearchSorted)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "test_searchsorted_op.py", line 45, in test_check_output
>     self.check_output(check_pir=True)
>   File "/home/aistudio/Paddle/test/legacy_test/op_test.py", line 2762, in check_output
>     check_auto_parallel=check_auto_parallel,
>   File "/home/aistudio/Paddle/test/legacy_test/op_test.py", line 2622, in check_output_with_place
>     pir_checker.check()
>   File "/home/aistudio/Paddle/test/legacy_test/op_test.py", line 2187, in check
>     self.calculate_output()
>   File "/home/aistudio/Paddle/test/legacy_test/op_test.py", line 2373, in calculate_output
>     pir_outs = self.op_test._calc_pir_output(place)
>   File "/home/aistudio/Paddle/test/legacy_test/op_test.py", line 1379, in _calc_pir_output
>     ret_tuple = self.python_api(*args)
>   File "/home/aistudio/.local/lib/python3.7/site-packages/paddle/tensor/search.py", line 1157, in searchsorted
>     out = helper.create_variable_for_type_inference(dtype=out_type)
>   File "/home/aistudio/.local/lib/python3.7/site-packages/paddle/base/layer_helper_base.py", line 476, in create_variable_for_type_inference
>     stop_gradient=stop_gradient,
>   File "/home/aistudio/.local/lib/python3.7/site-packages/paddle/base/framework.py", line 4223, in create_var
>     var = Variable(block=self, *args, **kwargs)
>   File "/home/aistudio/.local/lib/python3.7/site-packages/paddle/base/framework.py", line 1514, in __init__
>     self.desc.set_dtype(dtype)
> TypeError: set_dtype(): incompatible function arguments. The following argument types are supported:
>     1. (self: paddle.base.libpaddle.VarDesc, arg0: paddle::framework::proto::VarType_Type) -> None
> 
> Invoked with: <paddle.base.libpaddle.VarDesc object at 0x7f391f0a97f0>, <DataType.INT64: 9>
> ```

我看 ci 运行是正常的 😄 可以检查一下本地测试代码有没有同步 `python/paddle/tensor/search.py` 的更改

File: test/legacy_test/test_searchsorted_op.py
Comment (MarioLulab) [inline test/legacy_test/test_searchsorted_op.py:1]: 该文件下的 TestSearchSortedError 单测是否支持？如不支持请在 pr 描述里说明一下，方便我们记录问题并后续 fix

**Review by MarioLulab — COMMENTED** [review]
searchsorted api 在 test/legacy_test/test_zero_dim_tensor.py 里的 TestSundryAPIStatic.test_searchsorted 有相关单测，辛苦也适配一下 pir

**Comment by MarioLulab** [comment]
请问大佬这个 pr 最近还有空推进吗 ~  没有的话可以让内部研发接着做啦😃

**Comment by Olive-2019** [comment]
> 请问大佬这个 pr 最近还有空推进吗 ~ 没有的话可以让内部研发接着做啦😃


抱歉抱歉，最近期末事情太多了，可能没空推进了

**Comment by MarioLulab** [comment]
> > 请问大佬这个 pr 最近还有空推进吗 ~ 没有的话可以让内部研发接着做啦😃
> 
> 抱歉抱歉，最近期末事情太多了，可能没空推进了

好的没事~ 那我们就在你的代码上继续推进啦 🚀

**Comment by MarioLulab** [comment]
Done in https://github.com/PaddlePaddle/Paddle/pull/60014.
欢迎下次贡献~

**Pull request closed without merging**